nap/resources.py

```diff
@@ -212,3 +213,18 @@ def __repr__(self):
         val = '<{}: {}>'.format(self.__class__.__name__, self.__unicode__())
         return val
 
+
+class OpaqueFilterResourceModel(ResourceModel):
+    @classmethod
+    def update_resource_fields(cls, attributes):
+        """Dynamically setting Resource fields based off response fields"""
+        for attribute_name in attributes:
+            if not hasattr(cls, attribute_name):
+                # see DataModelMetaClass.__new__
+                temp_field = Field()
```

Comment:
This is future thinking, not something for this PR, but I wonder if it would be possible to specify field types so that we can take advantage of nap's automatic conversion of dates, decimals, lists, and dicts.